### PR TITLE
Add transit VPC access support

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/9187

1. Add routes into transit VPC if it is configured
2. Enable SSH access to bastions from transit VPC cidr

### CODE REVIEW

**Comment by corest** [comment]
This is deployed already for all China installations

File: modules/aws/vpc/vpc.tf [hunk 1]
```diff
@@ -139,6 +139,15 @@ resource "aws_route" "vpc_local_route" {
   depends_on             = [aws_route_table.cluster_vpc_public]
 }
 
+resource "aws_route" "vpc_transit_route" {
+  count = var.transit_vpc_cidr != "" ? length(var.subnets_elb) : 0
+
+  route_table_id         = element(aws_route_table.cluster_vpc_public.*.id, count.index)
```
Comment (kopiczko) [inline modules/aws/vpc/vpc.tf:145]: Does it mean all the traffic from `cluster_vpc_{public,private}` will go trough VPN? What is there?

File: modules/aws/vpc/variables.tf [hunk 2]
```diff
@@ -47,3 +47,17 @@ variable "with_public_access" {
   type    = bool
   default = false
 }
+
+### Access via transit VPC ###
+variable "vpc_vgw_id" {
```
[the remark this answers is not in the document]
Comment (corest) [reply modules/aws/vpc/variables.tf:52]: No, it is no transit. It is just a regular virtual private gateway, which can be attached to any VPC (in this case it is attached tp VPCs in China).

File: modules/aws/vpc/vpc.tf (hunk 1 above)
Comment (corest) [reply modules/aws/vpc/vpc.tf:145]: No, it means that we create this rule in all routing tables. And rule itself means that traffic, where the destination is `destination_cidr_block` at line below (in our case it is transit VPC CIDR) is routed via specified gateway

File: platforms/aws/giantnetes/main.tf [hunk 3]
```diff
@@ -105,7 +105,9 @@ module "vpc" {
   subnets_elb        = "${var.subnets_elb}"
   subnets_worker     = "${var.subnets_worker}"
   subnets_vault      = "${var.subnets_vault}"
+  transit_vpc_cidr = "${var.transit_vpc_cidr}"
```
[the remark this answers is not in the document]
Comment (corest) [reply platforms/aws/giantnetes/main.tf]: fixed

**Pull request merged**